webhooks/virtualmachinesnapshot/validation/virtualmachinesnapshot_validator.go

```diff
@@ -136,6 +143,30 @@ func (v validator) ValidateUpdate(ctx *pkgctx.WebhookRequestContext) admission.R
 	return common.BuildValidationResponse(ctx, nil, validationErrs, nil)
 }
 
+// validateVMNotVKSNode checks if the referenced VM is a VKS/TKG node and returns an error if it is.
+func (v validator) validateVMNotVKSNode(ctx *pkgctx.WebhookRequestContext, vmRef *vmopv1common.LocalObjectRef, namespace string) error {
```

Comment:
How about just in the snapshot controller?

Reply:
I think there's value in keeping this in the webhooks since this is going to be pretty deterministic.  Let me know if you disagree.